emulated_hue/api.py

```diff
@@ -627,14 +642,50 @@ async def __async_light_action(self, entity: dict, request_data: dict) -> None:
         if const.HUE_ATTR_TRANSITION in request_data:
             # Duration of the transition from the light to the new state
             # is given as a multiple of 100ms and defaults to 4 (400ms).
-            transitiontime = request_data[const.HUE_ATTR_TRANSITION] / 10
+            if request_data[const.HUE_ATTR_TRANSITION] * 100 <= throttle_ms:
+                transitiontime = throttle_ms / 1000
+            else:
+                transitiontime = request_data[const.HUE_ATTR_TRANSITION] / 10
             data[const.HASS_ATTR_TRANSITION] = transitiontime
         else:
-            data[const.HASS_ATTR_TRANSITION] = 0.4
+            data[const.HASS_ATTR_TRANSITION] = 0.4 if throttle_ms <= 400 else throttle_ms / 1000
 
         # execute service
         await self.hass.async_call_service(const.HASS_DOMAIN_LIGHT, service, data)
 
+    def __update_allowed(
+        self, light_id: str, light_data: dict, throttle_ms: int
+    ) -> bool:
+        """Minimalistic form of throttling, only allow updates to a light within a timespan."""
+
+        # check if data changed
+        # when not using udp no need to send same light command again
+        prev_data = self._prev_data.get(light_id, {})
+
+        # force to update if power state changed
+        if prev_data.get(const.HASS_STATE_ON, True) != light_data.get(const.HASS_STATE_ON, True):
+            return True
+        if prev_data.get(const.HUE_ATTR_BRI, 0) == light_data.get(const.HUE_ATTR_BRI, 0)\
```

Comment:
remove the backslash and put it all in brackets. black formatter will do this automagically for you btw.